### ISSUE TITLE
The schema for JsonSubTypes & JsonTypeInfo was not generated for simple kotlin project

### ISSUE THREAD

**Issue by pelenthium** [micronaut-projects/micronaut-openapi#1514]
### Expected Behavior

Follow  https://github.com/micronaut-projects/micronaut-openapi/issues/508 I expected that openapi specification will be generated something like this: 
`
...
schema:
              oneOf:
                - $ref: '#/components/schemas/ErrorResponse'
                - $ref: '#/components/schemas/SuccessResponse'
              discriminator:
                propertyName: type
                mapping:
                  ERROR: '#/components/schemas/ErrorResponse'
                  SUCCESS: '#/components/schemas/SuccessResponse'
`

### Actual Behaviour

Generated yml with follow content: 

```
openapi: 3.0.1
paths:
  /openapi:
    get:
      operationId: index
      responses:
        "200":
          description: index 200 response
          content:
            text/plain:
              schema:
                $ref: '#/components/schemas/BaseResponse'
components:
  schemas:
    BaseResponse:
      required:
      - id
      type: object
      properties:
        id:
          type: string
```

### Steps To Reproduce

1. Generate new project with command line tools `mn create-app --build=gradle_kotlin --jdk=21 --lang=kotlin --test=junit --features=openapi,redoc com.example.openapi`
2. build project and see in swagger.yml
 

### Environment Information

openjdk version "21.0.2" 2024-01-16

### Example Application

https://github.com/pelenthium/micronaut-openapi-subtype-issue

### Version

4.3.7

**Comment by altro3** [comment]
Already fixed in micronaiut-openapi `6.7.0` . Just use latest version like this:
```kotlin
    ksp("io.micronaut.openapi:micronaut-openapi:6.7.0!!")
    compileOnly("io.micronaut.openapi:micronaut-openapi-annotations:6.7.0!!")
```

**Comment by altro3** [comment]
Generated file for you example with micronaut-openapi 6.7.0:
![изображение](https://github.com/micronaut-projects/micronaut-openapi/assets/48474279/57ea501a-b944-4c42-b398-78103aba5abd)
